UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarning/BaseRNWTest.cs

```diff
@@ -0,0 +1,172 @@
+using UKHO.MaritimeSafetyInformation.Web.Services;
+using radioNavigationalWarningDto = UKHO.MaritimeSafetyInformation.Common.Models.RadioNavigationalWarning.DTO;
+
+namespace UKHO.MaritimeSafetyInformation.IntegrationTests.RadioNavigationalWarning
+{
+    public class BaseRNWTest
+    {
+        public readonly RadioNavigationalWarningsContext _fakeContext;
+        public IOptions<RadioNavigationalWarningConfiguration> _fakeRadioNavigationalWarningConfiguration;
+        public IHttpContextAccessor _fakeHttpContextAccessor;
+        public ILogger<RNWService> _fakeLoggerRnwService;
+
+        public BaseRNWTest()
+        {
+            DbContextOptionsBuilder<RadioNavigationalWarningsContext> builder = new DbContextOptionsBuilder<RadioNavigationalWarningsContext>()
+                                                                    .UseInMemoryDatabase("msi-in-db");
+            _fakeContext = new RadioNavigationalWarningsContext(builder.Options);
+            _fakeRadioNavigationalWarningConfiguration = A.Fake<IOptions<RadioNavigationalWarningConfiguration>>();
+            _fakeHttpContextAccessor = A.Fake<IHttpContextAccessor>();
+            _fakeLoggerRnwService = A.Fake<ILogger<RNWService>>();
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+        }
+
+        public async Task SeedRadioNavigationalWarnings(List<radioNavigationalWarningDto.RadioNavigationalWarning> radioNavigationalWarning)
+        {
+            _fakeContext.RadioNavigationalWarnings.AddRange(radioNavigationalWarning);
+            await _fakeContext.SaveChangesAsync();
+        }
+
+        public async Task SeedWarningType(List<radioNavigationalWarningDto.WarningType> warningTypes)
+        {
+            _fakeContext.WarningType.AddRange(warningTypes);
+            await _fakeContext.SaveChangesAsync();
+        }
+
+        public async Task DeSeedRadioNavigationalWarnings()
+        {
+            DbSet<radioNavigationalWarningDto.RadioNavigationalWarning> warnings = _fakeContext.RadioNavigationalWarnings;
+            _fakeContext.RadioNavigationalWarnings.RemoveRange(warnings);
+            await _fakeContext.SaveChangesAsync();
+        }
+
+        public async Task DeSeedWarningType()
+        {
+            DbSet<radioNavigationalWarningDto.WarningType> warningType = _fakeContext.WarningType;
+            _fakeContext.WarningType.RemoveRange(warningType);
+            await _fakeContext.SaveChangesAsync();
+        }
+
+        public static List<radioNavigationalWarningDto.RadioNavigationalWarning> GetFakeRadioNavigationalWarnings()
+        {
+            List<radioNavigationalWarningDto.RadioNavigationalWarning> radioNavigationalWarningList = new();
+
+            radioNavigationalWarningList.Add(new radioNavigationalWarningDto.RadioNavigationalWarning()
+            {
+                WarningType = WarningTypes.NAVAREA_1,
+                Reference = "RnwAdminListReferance",
```

Comment:
Please do a global replace of `Referance` with `Reference`.  e.g.:
```suggestion
                Reference = "RnwAdminListReference",
```

Reply:
Changes done as requested, Thank you.